src/dependencies.js

```diff
@@ -92,15 +92,22 @@ const getFileDependencies = async function(path, packageJson, state) {
 // `require()` statements can be either `require('moduleName')` or
 // `require(path)`
 const getImportDependencies = function(dependency, basedir, packageJson, state) {
-  if (LOCAL_IMPORT_REGEXP.test(dependency)) {
+  const moduleName = getModuleName(dependency)
+  if (LOCAL_IMPORT_REGEXP.test(dependency) || isTreeShakable(moduleName)) {
     return getLocalImportDependencies(dependency, basedir, packageJson, state)
   }
 
-  return getModuleDependencies(dependency, basedir, state, packageJson)
+  return getModuleDependencies(moduleName, basedir, state, packageJson)
 }
 
 const LOCAL_IMPORT_REGEXP = /^(\.|\/)/
 
+const isTreeShakable = function(moduleName) {
+  return TREE_SHAKABLE_MODULES.includes(moduleName)
+}
+
+const TREE_SHAKABLE_MODULES = ['next']
```

Comment:
Did we double-check that no part of `next` source code is either doing a dynamic require, or reading a file published to npm?

Reply:
I want to verify this works for the `Next.js` plugin first, then we need to think about other use cases.
Using specific paths would keep this scoped to `next-on-netlify` - good point

Reply:
See the list here:
https://github.com/netlify/zip-it-and-ship-it/pull/242/commits/940ca53cdb176c14b520abbcb66d175b7aa13a67#diff-d960073cd7b388518947c4e516460d086b7c49fdb95b3de6b16c3edfd002a8ccR108
based on https://github.com/erezrokah/erez-test-next-bundling (Next.js 10).

In `serverless` mode Next.js inlines all dependencies except `@ampproject/toolbox-optimizer`:
https://github.com/vercel/next.js/blob/4bca939ff72c83fbedae2cba91317e91b804c8d3/packages/next/build/webpack-config.ts#L723

In `experimental-serverless-trace` it uses the following logic:
https://github.com/vercel/next.js/blob/4bca939ff72c83fbedae2cba91317e91b804c8d3/packages/next/build/webpack-config.ts#L562 (this might imply we can do the same for the entire `next` package)

Without this fix functions sizes for https://github.com/erezrokah/erez-test-next-bundling are:
```
25M Dec  9 16:47 next_api_hello.zip
25M Dec  9 16:47 next_test.zip
```

With this fix functions sizes for https://github.com/erezrokah/erez-test-next-bundling are:
```
345K Dec  9 16:42 next_api_hello.zip
5.0M Dec  9 16:42 next_test.zip
```